### ISSUE TITLE
Config with object that `_takes_rng`

### ISSUE THREAD

**Issue by esheldon** [GalSim-developers/GalSim#820]
I tried to test using a config with `RandomWalk`, which has `_takes_rng=True` set.

I initially created the following configuration
```python
    gal_config = {
        'type':'RandomWalk',
        'npoints':100,
        'half_light_radius':2.0,
    }
    config={
        'gal':gal_config,
    }
```

An exception is raised because the `rng` is missing.  This surprised me, because `rng` is an optional parameter
```python
Traceback (most recent call last):
  File "test_randwalk.py", line 253, in <module>
    test_randwalk_config()
  File "/home/esheldon/git/GalSim/tests/galsim_test_helpers.py", line 456, in f2
    result = f(*args, **kwargs)
  File "test_randwalk.py", line 199, in test_randwalk_config
    rw_from_config = galsim.config.BuildGSObject(config, 'gal')
  File "/home/esheldon/anaconda/lib/python2.7/site-packages/galsim/config/gsobject.py", line 182, in BuildGSObject
    gsobject, safe = _BuildSimple(param, base, ignore, gsparams, logger)
  File "/home/esheldon/anaconda/lib/python2.7/site-packages/galsim/config/gsobject.py", line 256, in _BuildSimple
    raise ValueError("No base['rng'] available for type = %s"%type_name)
ValueError: No base['rng'] available for type = RandomWalk

```

Note `flux` is not being set above, it is optional, yet no error is raised.

This goes away if I instead add the `rng` to the _outer_ config
```python
    gal_config = {
        'type':'RandomWalk',
        'npoints':100,
        'half_light_radius':2.0,
        'flux':np.pi,
    }
    config={
        'gal':gal_config,
        'rng':galsim.BaseDeviate(),
    }
```

So I have two questions:  

1. Should the `rng` be dealt with like other optional parameters?
2. Must the rng always be at the outer config?

**Comment by rmandelb** [comment]
I think that the ```test_config_gsobject.py``` test for ```RealGalaxy```  (another one with ```takes_rng=True```), ```test_realgalaxy```, suggests an answer.  You don't need to provide an ```rng``` for a RealGalaxy/RandomWalk, but config does need to have some ```rng``` set otherwise in the outer config.  Hopefully Mike will comment if I've misunderstood here.

**Comment by rmjarvis** [comment]
Right.  Within the config stuff, we typically want our scripts to be repeatable.  So that means always providing an explicit rng rather than let the object make its own from /dev/urandom.  

If the user doesn't specify a seed, then the config layer will make the default rng and pass that in.  So when actually running a script, this should never be a problem.

However, in tests, where you only run a portion of the config functionality, you sometimes need to add `config['rng'] = galsim.BaseDeviate(123)`.  cf. tests in test_config_value.py and test_config_gsobject.py for examples.

**Comment by rmjarvis** [comment]
Merged via #838
